docs/decisions/0008-migrating-to-new-learner-credit.md

```diff
@@ -0,0 +1,159 @@
+  * `maxDiscount`. Maximum total spend allowed.
+  * `maxGlobalApplications`. Maximum total number of enrollments allowed.
+  * `remainingBalance`. The total available balance remaining to be spent by all learners.
+  * `remainingBalanceforUser`. The available balance remaining specifically for the authenticated user.
+  * `isLowOnBalance`. Determined by [`isOfferLowOnBalance`](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/5034f5ab170589a923c223cfe238112eff48f5c4/src/components/enterprise-user-subsidy/enterprise-offers/data/utils.js#L32).
+  * `isOutOfBalance`. Determined by [`isOfferOutOfBalance`](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/5034f5ab170589a923c223cfe238112eff48f5c4/src/components/enterprise-user-subsidy/enterprise-offers/data/utils.js#L51).
+* `canEnrollWithEnterpriseOffers`. Boolean representing whether there is at least one enterprise learner credit available.
+* `hasLowEnterpriseOffersBalance`. Boolean representing whether any of the `enterpriseOffers` are low on balance.
+* `hasNoEnterpriseOffersBalance`. Boolean representing whether all `enterpriseOffers` have no balance remaining.
+
+#### Dashboard
+
+This page route relies on the learner credit data in order to display messaging to learners informing them of having learner credit available to spend. The UI component rendered in the dashboard page's sidebar is the [`EnterpriseOffersSummaryCard`](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/5034f5ab170589a923c223cfe238112eff48f5c4/src/components/dashboard/sidebar/EnterpriseOffersSummaryCard.jsx).
+
+This component gets rendered within the [`SubsidiesSummary`](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/5034f5ab170589a923c223cfe238112eff48f5c4/src/components/dashboard/sidebar/SubsidiesSummary.jsx#L10) component, which renders similar UI components for other subsidies (e.g., subscription licenses).
+
+The `SubsidiesSummary` component gets its data about available learner credit from the `UserSubsidyContext`.
+
+`EnterpriseOffersSummaryCard` either displays a generic message for max global spend or a user-specific message for max user spend remaining. It does not currently support any enrollment limits (e.g., `maxGlobalApplications`). In the case of max user spend remaining, we sum all remaining user balance for all available learner credit. The expiration date shown is from the learner credit that expires first.
+
+#### Search
+This page route similarly relies on the data provided by the `UserSubsidyContext` in order to display an alert to inform learners of low/no balance remaining. The rationale for this alert in the UX is to help proactively make learners are aware they may not have enough funds available to cover the cost of all content returned in the search results.
```

Comment:
This context about alerting on low or no balance remaining seems uncovered by the decisions section.  What should we do when new learner credit subsidies have low or no balance?  How should low balance be defined, since we're returning raw balance amounts instead of booleans now?

Reply:
Great questions! The context about the alerting for low/no balance remaining was written before I realized we don't actually need to focus on it for the initial EMET release given we're only tackling use cases where the learners are coming from an external LMS. Given that, the dashboard/search pages are not accessible to learners coming from an LMS so the implications for the dashboard/search page are not being accounted for in the initial EMET release. That's probably why the "Decisions" section doesn't really cover it.

That said, you bring up a great point around whether the API should be making the determination of "low" or "no" balance remaining. The existing business logic in the frontend for this are capturing in the following functions:
* `isOfferLowOnBalance` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/ddad30269725b056ea4c6f2f933332b722b1002a/src/components/enterprise-user-subsidy/enterprise-offers/data/utils.js#L32))
* `isOfferOutOfBalance` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/ddad30269725b056ea4c6f2f933332b722b1002a/src/components/enterprise-user-subsidy/enterprise-offers/data/utils.js#L51))

I'd be in favor of eventually getting this in to the API layer to further remove business logic from the frontend, but I don't think we necessarily need to take action on it until EMET has a requirement to support non-LMS use cases, too.

Reply:
Thanks!  My only suggestion then would be to add a placeholder section/paragraph in decisions to explicitly defer decisions about the balance

Reply:
Agreed. Will update!